doc/services/storage/zms/zms.rst

```diff
@@ -0,0 +1,418 @@
+4.720.000 minutes (~9 years).
+
+To make a more general formula we must first compute the effective used size in ZMS by our
+typical set of data.
+For id/data pair with data <= 8 bytes, effective_size is 16 bytes
+For id/data pair with data > 8 bytes, effective_size is 16 bytes + sizeof(data)
+Let's suppose that total_effective_size is the total size of the set of data that is written in
+the storage and that the partition is well dimensioned (double of the effective size) to avoid
+having the garbage collector moving blocks all the time.
+
+The expected life of the device in minutes is computed as ::
+
+(SECTOR_EFFECTIVE_SIZE * SECTOR_NUMBER * MAX_NUM_WRITES) / (TOTAL_EFFECTIVE_SIZE * WR_MIN)
+
+Where:
+
+``SECTOR_EFFECTIVE_SIZE``: is the size sector - header_size(80 bytes)
+
+``SECTOR_NUMBER``: is the number of sectors
+
+``MAX_NUM_WRITES``: is the life expectancy of the storage device in number of writes
+
+``TOTAL_EFFECTIVE_SIZE``: Total effective size of the set of written data
+
+``WR_MIN``: Number of writes of the set of data per minute
+
+ZMS Features
+************
+ZMS has introduced many features compared to existing storage system like NVS and will evolve
+from its initial version to include more features that satisfies new technologies requirements
+such as low latency and bigger storage space.
+
+Existing features
+=================
+Version1
+--------
+- Supports non erasable devices (only one write operation to erase a sector)
+- Supports larger partition size and sector size (64 bits address space)
+- Supports larger IDs (32 bits) to store ID/Value pairs
+- Small sized data ( <= 8 bytes) are stored in the ATE itself
+- Built-in Data CRC32 (included in the ATE)
+- Versionning of ZMS (to handle future evolution)
+- Supports larger write-block-size (Only for platforms that need this)
+- Enables customization of sector count (Still under experimentation)
+
+Future Extensions
+=================
+
+- Add multiple format ATE support to be able to use ZMS with different ATE formats that satisfies
+  requirements from application
+- Add the possibility to skip garbage collector for some application usage where ID/value pairs
+  are written periodically and do not exceed half of the partition size (ther is always an old
+  entry with the same ID).
+- Divide IDs into namespaces and allocate IDs on demand from application to handle collisions
+  between IDs used by different subsystems or samples.
+- Add the possibility to retrieve the wear out value of the device based on the cycle count value
+- Add a recovery function that can recover a storage partition if something went wrong
+- Add a library/application to allow migration from NVS entries to ZMS entries
+- Add the possibility to force formatting the storage partition to the ZMS format if something
+  went wrong when mounting the storage.
+
+ZMS vs NVS, what should I choose ?
+==================================
+
```

Comment:
This section describes ZMS in the wider context of storage systems in Zephyr (not full filesystems, but simpler, non-hierarchical ones).
Today Zephyr includes at least two other systems that are somewhat comparable in scope and functionality: :ref:`NVS <nvs_api>` and :ref:`FCB <fcb_api>`.
Which one to use in your application will depend on your needs and the hardware you are using, and this section provides information to help make a choice.